app/models/hub_edos/student.rb

```diff
@@ -0,0 +1,32 @@
+module HubEdos
+  class Student
+    attr_accessor :terms_in_attendance, :academic_level
+
+    def initialize(uid)
+      @uid = uid
+      if uid.present?
+        @student_hash = HubEdos::V2::Student.new(user_id: uid).get
+      end
+    end
+
+    def get_max_terms_in_attendance
+      if statuses = @student_hash[:feed]['academicStatuses']
+        return statuses.collect {|s| s['termsInAttendance']}.sort.last
+      end
+    end
+
+    def get_student_academic_level
+      current_registration = get_current_registration
```

Comment:
If you memoize `get_current_registration` you can use it within this method without assigning the temporary variable.

Reply:
Yes. I thought of doing that, but decided to go this way given that there is only one caller currently.